sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpDataBody.java

```diff
@@ -3,26 +3,69 @@
 
 package com.azure.core.amqp.models;
 
+import com.azure.core.experimental.util.BinaryData;
 import com.azure.core.util.IterableStream;
+import com.azure.core.util.logging.ClientLogger;
 
+import java.util.Collections;
+import java.util.List;
 import java.util.Objects;
 
 /**
  * This is amqp message body which represents {@link AmqpBodyType#DATA} type.
  */
 public final class AmqpDataBody implements AmqpMessageBody {
-    private final IterableStream<byte[]> data;
+    private final ClientLogger logger = new ClientLogger(AmqpDataBody.class);
+    private final BinaryData binaryData;
 
     /**
-     * Creates instance of {@link AmqpDataBody} with given {@link Iterable} of byte array.
+     * Creates instance of {@link AmqpDataBody} with given {@link Iterable} of byte array. As of now,  We support
+     * only one element in the given iterable.
      *
      * @param data to be set on amqp body.
      *
      * @throws NullPointerException if {@code data} is null.
+     * @throws IllegalArgumentException if elements in 'data' is zero or greater than one.
      */
     public AmqpDataBody(Iterable<byte[]> data) {
         Objects.requireNonNull(data, "'data' cannot be null.");
-        this.data = new IterableStream<>(data);
+        BinaryData payload = null;
+        for (byte[] bytes : data) {
+            if (payload != null) {
+                throw logger.logExceptionAsError(new IllegalArgumentException(
+                    "Only one instance of byte array is allowed in 'data'."));
+            }
+            payload = BinaryData.fromBytes(bytes);
+        }
+
+        if (payload == null) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("'data' can not be empty."));
+        } else {
+            this.binaryData = payload;
+        }
+    }
+
+    /**
+     * Creates instance of {@link AmqpDataBody} with given {@link List} of {@link BinaryData}. As of now,  We support
+     * only one element in the given collection.
+     *
+     * @param data to be set on amqp body.
+     *
+     * @throws NullPointerException if {@code data} is null.
+     * @throws IllegalArgumentException if size of 'data' is zero or greater than one.
+     */
+    public AmqpDataBody(List<BinaryData> data) {
```

Comment:
This is not what we want to do - this is where the breaking change that was previously done is hugely regrettable. We don't want constructors that take Iterable and List without any clear reason why (other than we stuffed up in a previous release).